src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackCypressReportForBuild.java

```diff
@@ -0,0 +1,162 @@
+
+public class BrowserStackCypressReportForBuild extends AbstractBrowserStackCypressReportForBuild {
+    private static PrintStream logger;
+    private final String buildName;
+    private final transient JSONObject result;
+    private final Map<String, String> resultAggregation;
+    private final ProjectType projectType;
+    // to make them available in jelly
+    private final String passedConst = Constants.SessionStatus.PASSED;
+    private final String failedConst = Constants.SessionStatus.FAILED;
+    private final transient PluginsTracker tracker;
+    private final boolean pipelineStatus;
+
+    public BrowserStackCypressReportForBuild(final Run<?, ?> build,
+                                             final ProjectType projectType,
+                                             final String buildName,
+                                             final PrintStream logger,
+                                             final PluginsTracker tracker,
+                                             final boolean pipelineStatus) {
+        super();
+        setBuild(build);
+        this.buildName = buildName;
+        this.result = new JSONObject();
+        this.resultAggregation = new HashMap<>();
+        this.projectType = projectType;
+        this.logger = logger;
+        this.tracker = tracker;
+        this.pipelineStatus = pipelineStatus;
+    }
+
+
+    public JSONObject getCypressMatrix(String filepath) {
+        JSONObject report = null;
+        final String reportJSONPath = filepath + "/results/browserstack-cypress-report.json";
+
+        try {
+            InputStream is = new FileInputStream(reportJSONPath);
+            String jsonTxt = IOUtils.toString(is, "UTF-8");
+            report = new JSONObject(jsonTxt);
+        } catch (FileNotFoundException e) {
+            logError(logger, "No BrowserStackBuildAction found");
+            tracker.sendError("BrowserStack Cypress Report Not Found", pipelineStatus, "ReportGeneration");
+        } catch (IOException e) {
+            logError(logger, "There was a problem while reading report files");
+            tracker.sendError(e.toString(), pipelineStatus, "ReportGeneration");
+        }
+        return report;
+    }
+
+    public boolean generateBrowserStackCypressReport(String workspace) {
+        if (result.length() == 0) {
+            JSONObject matrix = getCypressMatrix(workspace);
+
+            if(matrix == null) {
+                return false;
+            }
+
+            String buildNameWithBuildNumber = matrix.getString("build_name");
+            String buildNameWithoutBuildNumber = buildNameWithBuildNumber.substring(0, buildNameWithBuildNumber.lastIndexOf(": "));
+
+            if (!buildNameWithoutBuildNumber.equalsIgnoreCase(this.buildName)) {
+                logError(logger, "BrowserStack Cypress Report not generated, build name mismatch.");
+                tracker.sendError("Report not generated", pipelineStatus, "ReportGeneration");
```

Comment:
We probably would need to separate the `tags` here with the existing events. Maybe add `cypress` as prefix or suffix to it...

Reply:
Do you mean something like 
`tracker.sendError("Report not generated", pipelineStatus, "CypressReportGeneration");`

Reply:
Yep. To differentiate it from the rest...

Reply:
Can be resolved.